_episodes/08-connected-components.md

```diff
@@ -261,156 +319,388 @@ max: 11
+documentation to learn about other properties available.
+
+We can get a list of areas of the labeled objects as follows:
 
-> ## Plot a histogram of the object area distribution (15 min)
->
-> In the previous exercise we wrote the `CCA-count.py` program and explored how the object count changed with the parameters.
-> We had a hard time making the script print out the right number of objects.
-> In order to get closer to a solution to this problem, we want to look at the distribution of the object areas.
-> Calculate the object properties using `skimage.measure.regionprops` and [generate a histogram]({{ page.root }}/05-creating-histograms).
->
-> Make a copy of the `CCA.py` script and modify it to also produce a plot of the histogram of the object area.
+~~~
+# compute object features and extract object areas
+object_features = skimage.measure.regionprops(labeled_image)
+object_areas = [objf["area"] for objf in object_features]
+object_areas
+~~~
+{: .language-python}
+
+This will produce the output
+~~~
+[318542, 1, 523204, 496613, 517331, 143, 256215, 1, 68, 338784, 265755]
+~~~
+{: .output}
+
+> ## Plot a histogram of the object area distribution (10 min)
 >
-> What does the histogram tell you about the objects?
+> Similar to how we determined a "good" threshold in the
+> [Thresholding]({{ page.root }}/07-thresholding) episode, it is often
+> helpful to inspect the histogram of an object property. For example,
+> we want to look at the distribution of the object areas.
 >
-> Hint: Try to generate a list of object areas first.
+> 1. Create and examine a [histogram]({{ page.root }}/05-creating-histograms)
+> of the object areas obtained with `skimage.measure.regionprops`.
+> 2. What does the histogram tell you about the objects?
 >
 > > ## Solution
 > >
+> > The histogram can be plotted with
 > > ~~~
-> > # add the import for pyplot
-> > from matplotlib import pyplot as plt
-> >
-> > # compute object features and extrac object areas
-> > object_features = skimage.measure.regionprops(labeled_image)
-> > object_areas = [objf["area"] for objf in object_features]
 > > plt.hist(object_areas)
-> > plt.show()
 > > ~~~
 > > {: .language-python}
+> >
+> > ![Histogram of object areas](../../fig/07-areas-histogram.png)
+> >
+> > The histogram shows the number of objects (vertical axis) whose
+> > area is within a certain range (horizontal axis). The height of
+> > the bars in the histogram indicates the prevalence of objects with
+> > a certain area. The whole histogram tells us about the
+> > distribution of object sizes in the image. It is often possible to
+> > identify gaps between groups of bars (or peaks if we draw the
+> > histogram as a continuous curve) that tell us about certain groups
+> > in the image.
+> >
+> > In this example, we can see that there are four small objects that
+> > contain less than 50000 pixels. Then there is a group of four
+> > (1+1+2) objects in the range between 20000 and 400000, and three
```

Comment:
```suggestion
> > (1+1+2) objects in the range between 200000 and 400000, and three
```

Reply:
Addressed in a82d659bb036937700f773606fb2e7b03f2e7a3a.